src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/FilterMetaRepository.cs

```diff
@@ -79,17 +82,19 @@ await optionTable
                 .InsertWhenNotMatched()
                 .MergeAsync(cancellationToken);
 
-            var startIndex = await publicDataDbContext.FilterOptionMetaLinks.CountAsync(token: cancellationToken) + 1;
+            var startIndex = (await publicDataDbContext.Database
+                    .SqlQuery<int>($"SELECT nextval('public.\"FilterOptionMetaLink_Id_seq\"')")
+                    .ToListAsync(cancellationToken: cancellationToken))
+                .First();
```

Comment:
Can this be written as the following:

```cs
var startIndex = await publicDataDbContext.Database
    .SqlQuery<int>($"SELECT nextval('public.\"FilterOptionMetaLink_Id_seq\"')")
    .FirstAsync(cancellationToken: cancellationToken);
```

Would also be tempted to abstract this into a method on `PublicDataDbContext`:

```cs
public Task<int> NextSequenceValue(string sequenceName, CancellationToken cancellationToken = default)
    => Database
        .SqlQuery<int>($"SELECT nextval('public.\"{sequenceName}\"')")
        .FirstAsync(cancellationToken);
```

This can then be re-used for importing location options and the seeder

Reply:
The reason for using `First` after `ToListAsync` and not `FirstAsync` is because that was giving me the error 

```column t.Value does not exist```

I've just realised it's documented that with `SqlQuery`/`SqlQueryRaw` the output column in the `SELECT` must be named `Value` and that it's case sensitive. In Postgres this means we need to wrap `Value` in quotes.

I've abstracted out `NextSequenceValue` and `SetSequenceValue` as suggested which makes things quite a bit neater, thanks. Unfortunately that string interpolation doesn't work well with the function calls, just like we've seen elsewhere executing DDL commands.

This is the neatest syntax that I've managed to come up with to get this working so far:

```csharp
    [SuppressMessage("Security", "EF1002:Risk of vulnerability to SQL injection.")]
    public Task<int> NextSequenceValue(string sequenceName, CancellationToken cancellationToken = default)
    {
        return Database.SqlQueryRaw<int>($"""
                                          SELECT nextval('public."{sequenceName}"') AS "Value"
                                          """)
            .FirstAsync(cancellationToken);
    }

    [SuppressMessage("Security", "EF1002:Risk of vulnerability to SQL injection.")]
    public Task<int> SetSequenceValue(string sequenceName, long value, CancellationToken cancellationToken = default)
    {
        return Database.SqlQueryRaw<int>($"""
                                          SELECT setval('public."{sequenceName}"', {value}) AS "Value"
                                          """)
            .FirstAsync(cancellationToken);
    }
```

The method `SetSequenceValue` doesn't really need to return a value but I've put it in anyway because it does return a value from the database and both methods then look more consistent.